testing/integration/kiosk-proxy.nix

```diff
@@ -0,0 +1,72 @@
+let
+  pkgs = import ../../pkgs { };
+  serverPort = 8080;
+  proxyPort = 8888;
+  kiosk = import ../../kiosk {
+    pkgs = pkgs;
+    system_name = "PlayOS";
+    system_version = "1.0.0";
+  };
+  toString = builtins.toString;
+in
+pkgs.nixosTest {
+  name = "proxy-test";
+
+  nodes = {
+    client = { config, ... }: {
+      imports = [
+        (pkgs.importFromNixos "tests/common/user-account.nix")
+        (pkgs.importFromNixos "tests/common/x11.nix")
+      ];
+
+      # Override is needed to enable in test VM, see connman tests:
+      # https://github.com/NixOS/nixpkgs/blob/1772251828be641110eb9a47ef530a1252ba211e/nixos/tests/connman.nix#L47-L52
+      services.connman.enable = pkgs.lib.mkOverride 0 true;
+
+      # We need a graphical environment and regular user for the kiosk browser
+      services.xserver = {
+        enable = true;
+      };
+      test-support.displayManager.auto.user = "alice";
+
+      environment.systemPackages = [
+        pkgs.connman
+        kiosk
+      ];
+    };
+  };
+
+  testScript = ''
+    start_all()
+
+    # Wait for X11 and connman, required by kiosk
+    client.wait_for_x()
+    client.wait_for_unit("connman.service")
+
+    with subtest('kiosk-browser uses configured proxy'):
+      service_name = client.succeed("connmanctl services | head -1 | awk '{print $3}'").strip(' \t\n\r')
+      client.succeed(f"connmanctl config {service_name} proxy manual http://user:p4ssw0rd@theproxy:${toString proxyPort}")
+
+      kiosk_result = client.execute(
+        'su - alice -c "kiosk-browser http://thecloud:${toString serverPort} http://foo.xyz" 2>&1',
+        check_return=False,
+        check_output=True,
+        timeout=10
+      )
+
+      # Ideally here we would check if starting the kiosk resulted in a request
```

Comment:
Accidentally spotted this, but this is quite easily doable by e.g. setting up tinyproxy on the other VM and checking tinyproxy's logs: https://github.com/dividat/playos/blob/e2a4379fafd085c7ec9ad70df81c1e4e8a70ba9f/testing/integration/controller-proxy.nix#L45-L60

Reply:
Ah yeah, thanks for the input.

The problem as I recall had not been inspecting the proxy, but getting the playos peer to resolve the names. I don't recall all details now, but think you may have found a solution in your recent tests, so it's time to look at this again.

Reply:
Yes, your technique applies, I added back the full test based on that. :pray: